### PR TITLE
Adds referrer feature.

### PR DESCRIPTION
**Pull request by pennello**
[OTPL-1961][1]

[1]: https://opentable.atlassian.net/browse/OTPL-1961

### CODE REVIEW

**Comment by pennello** [comment]
Travis checks failing because we depend on otj-core now (which is private).

File: client/src/main/java/com/opentable/jaxrs/referrer/ClientReferrerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * this does not currently include the {@code Referer} header.
+ *
+ * <p>
+ * If not using Spring, you would construct and set up the {@link ClientReferrerFeature} and
+ * {@link ClientReferrerFilter} yourself.
+ */
+@Configuration
+@Import(ClientReferrerFeature.class)
+public class ClientReferrerConfiguration {
+    static final String SERVICE_HEADER_NAME = "OT-Referring-Service";
```
Comment (mikebell90) [inline client/src/main/java/com/opentable/jaxrs/referrer/ClientReferrerConfiguration.java]: Make these public so people that don't directly use this feature  can at least have the string constant?

Comment (stevenschlansker) [inline client/src/main/java/com/opentable/jaxrs/referrer/ClientReferrerConfiguration.java]: Note that this duplicates https://github.com/opentable/otj-httpheaders/blob/master/src/main/java/com/opentable/httpheaders/OTHeaders.java so we probably should sort that out

Comment (pennello) [reply client/src/main/java/com/opentable/jaxrs/referrer/ClientReferrerConfiguration.java]: Ok, good call.  I updated it to use the common component in b1f2a14a174a420afd9e9b00a2786bef7529ee62.

Comment (pennello) [reply client/src/main/java/com/opentable/jaxrs/referrer/ClientReferrerConfiguration.java]: See b1f2a14a174a420afd9e9b00a2786bef7529ee62.

File: client/src/test/java/com/opentable/jaxrs/referrer/package-info.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Find referrer feature tests in otj-server.
+ */
+package com.opentable.jaxrs.referrer;
```
Comment (pennello) [inline client/src/test/java/com/opentable/jaxrs/referrer/package-info.java]: cf. https://github.com/opentable/otj-server/pull/38

**Comment by pennello** [comment]
Thanks guys!

**Pull request merged**